### PR TITLE
Remove default parameters for cudf::strings::detail functions

### PR DESCRIPTION
**Pull request by davidwendt**
## Description
Removes default parameters from the `cudf::strings::detail` functions. Most of these were unintentional the rest were for allowing for the default memory-resource which was easily fixed. Most of the detail functions are not used outside of strings and the default parameters were not actually necessary there.

Hopefully this will help with #11967 

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by bdice — APPROVED** [review]

File: cpp/src/strings/filling/fill.cu [hunk 1]
```diff
@@ -19,7 +19,8 @@
 #include <cudf/detail/valid_if.cuh>
 #include <cudf/null_mask.hpp>
 #include <cudf/scalar/scalar_device_view.cuh>
-#include <cudf/strings/combine.hpp>
+//#include <cudf/strings/combine.hpp>
```
Comment (PointKernel) [inline cpp/src/strings/filling/fill.cu:22]:
```suggestion
```
Can this be removed?

**Review by PointKernel — APPROVED** [review]
One minor question about the comment-out header inclusion.

LGTM. Thanks!

File: cpp/src/strings/filling/fill.cu (hunk 1 above)
Comment (bdice) [reply cpp/src/strings/filling/fill.cu:22]: Agreed. I'll commit this so that CI can re-run, since we have two approvals.

Comment (davidwendt) [reply cpp/src/strings/filling/fill.cu]: Oops. Yes. Thanks.

**Comment by davidwendt** [comment]
rerun tests

**Comment by davidwendt** [comment]
@gpucibot merge

**Pull request merged**